File: pocketr-java.api/src/main/java/com/decrux/pocketr/api/entities/dtos/AccountDto.java
package com.decrux.pocketr.api.entities.dtos;

import java.time.Instant;
import java.util.UUID;

public record AccountDto(
    UUID id,
    long ownerUserId,
    String name,
    String type,
    String currency,
    Instant createdAt
) {
    public UUID getId() {
        return id;
    }

    public long getOwnerUserId() {
        return ownerUserId;
    }

    public String getName() {
        return name;
    }

    public String getType() {
        return type;
    }

    public String getCurrency() {
        return currency;
    }

    public Instant getCreatedAt() {
        return createdAt;
    }
}

File: pocketr-api/src/main/kotlin/com/decrux/pocketr/api/entities/dtos/AccountDto.kt
package com.decrux.pocketr.api.entities.dtos

import java.time.Instant
import java.util.UUID

data class AccountDto(
    val id: UUID,
    val ownerUserId: Long,
    val name: String,
    val type: String,
    val currency: String,
    val createdAt: Instant,
)
